src/hooks/useQuery.js

```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+
+const useQuery = () => {
```

Comment:
I liked how you came up with this to get search params in custom components instead of using useLocation everywhere

Just a note, there is already a popular [useQuery](https://tanstack.com/query/v4/docs/react/reference/useQuery) hook from [TanStack Query](https://tanstack.com/) and this hook's name is the same as that one. Maybe changing the name of the hook to something like useSearchParams or something similar would prevent others from confusing this custom hook with the other one. This is just a suggestion and really up to you to change the name or not :)

Reply:
Thanks for notifying. Update to `useURLSearchParams` since `useSearchParams` is existing hook in `react-router-dom` :)